### DISCUSSION TITLE
Any plans to modularize (Java 9 modules) webauthn4j?

### DISCUSSION THREAD

**Discussion by hrstoyanov** [webauthn4j/webauthn4j#961] [Q&A]
Hard to work with it in fully modularized app ...

**Comment by ynojima** [comment]
Thank you for heads up. We don't have any plans at the moment, but we'd like to consider it in the long term.

**Reply by hrstoyanov** [reply]
Thank you @ynojima , but without any modularization, webauthn4j creates issues for modern java applications. Can you, please consider a short term solution - you just add these  entries to the META-INF/MANIFEST in each jar:

```
 Automatic-Module-Name: com.webauthn4j.core
 Automatic-Module-Name: com.webauthn4j.util
...
```

In Gradle:
```groovy
tasks.jar {
    manifest {
        attributes(
            "Automatic-Module-Name" to "com.example.mymodule"
        )
    }
}
```

This way, you reserve and claim the module names now, so that the compiler does not try to generate them from the jar file name (bad idea). 

And later, when you have the time, you could still add the proper **module-info.java** descriptors inside the jars.

**Reply by ynojima** [reply]
Addresses this with thw following pull-request: https://github.com/webauthn4j/webauthn4j/pull/988
